src/hyp3_isce2/burst.py

```diff
@@ -401,38 +401,54 @@ def validate_bursts(reference: str | list[str], secondary: str | list[str]) -> N
     """
     if isinstance(reference, str):
         reference = [reference]
+
     if isinstance(secondary, str):
         secondary = [secondary]
 
     if len(reference) < 1 or len(secondary) < 1:
-        raise ValueError('Must include at least 1 reference and 1 secondary burst')
+        raise ValueError('Must include at least 1 reference scene and 1 secondary scene')
+
     if len(reference) != len(secondary):
-        raise ValueError('Must have the same number of reference and secondary bursts')
+        raise ValueError(
+            f'Must provide the same number of reference and secondary scenes, got {len(reference)} reference and {len(secondary)} secondary'
+        )
+
+    ref_num_swath_pol = [_num_swath_pol(g) for g in reference]
+    sec_num_swath_pol = [_num_swath_pol(g) for g in secondary]
 
-    ref_num_swath_pol = sorted(g.split('_')[1] + '_' + g.split('_')[2] + '_' + g.split('_')[4] for g in reference)
-    sec_num_swath_pol = sorted(g.split('_')[1] + '_' + g.split('_')[2] + '_' + g.split('_')[4] for g in secondary)
-    if ref_num_swath_pol != sec_num_swath_pol:
-        msg = 'The reference and secondary burst ID sets do not match.\n'
-        msg += f'    Reference IDs: {ref_num_swath_pol}\n'
-        msg += f'    Secondary IDs: {sec_num_swath_pol}'
-        raise ValueError(msg)
+    for i in range(len(reference)):
+        if ref_num_swath_pol[i] != sec_num_swath_pol[i]:
+            raise ValueError(
+                f'Number + swath + polarization identifier does not match for reference scene {reference[i]} and secondary scene {secondary[i]}'
+            )
 
-    pols = list(set(g.split('_')[4] for g in reference + secondary))
+    pols = list(set(g.split('_')[4] for g in reference))
```

Comment:
You need to create `pols` using both reference and secondary. If you only use `reference`, you're not checking if the reference/secondary scenes share one single polarization.

Reply:
We've already checked that number+swath+pol is the same for each ref/sec pair, so we should only need to get the set of reference pols here, right?

Reply:
Yep, you're right